frontend/packages/kubevirt-plugin/src/statuses/template/template-source-status.ts

```diff
@@ -42,13 +42,40 @@ export const getTemplateSourceStatus: GetTemplateSourceStatus = ({
   pods,
   pvcs,
   dataVolumes,
+  dataSources,
 }) => {
   if (isCommonTemplate(template)) {
     const baseImageName = getParameterValue(template, TEMPLATE_BASE_IMAGE_NAME_PARAMETER);
     const baseImageNs = getParameterValue(template, TEMPLATE_BASE_IMAGE_NAMESPACE_PARAMETER);
     if (!baseImageName || !baseImageNs) {
       return null;
     }
+
+    const matchedDataSource = dataSources?.find(({ metadata }) => metadata?.name === baseImageName);
+    const matchedDataSourcePVC = pvcs?.find(
+      ({ metadata }) =>
+        metadata?.name === matchedDataSource?.spec?.source?.pvc?.name &&
+        metadata?.namespace === matchedDataSource?.spec?.source?.pvc?.namespace,
+    );
```

Comment:
```.find(
      ({ metadata }) =>
        metadata?.name === matchedDataSource?.spec?.source?.pvc?.name &&
        metadata?.namespace === matchedDataSource?.spec?.source?.pvc?.namespace,
    );
```
this function is used twice, extract it to `findByDataSource`

Reply:
extracted :)